polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -3314,10 +3316,19 @@ void MLIRASTConsumer::createMLIRParametersTypes(
     parmTypes.push_back(mlirThisType);
   }
 
+  const bool isKernel = FD.hasAttr<SYCLKernelAttr>();
```

Comment:
`SPIR_FUNC`s should also be included here.

Reply:
This PR concerns itself on passing pointer arguments 'byval' for only device functions that can be called from the host (that is only SPIR kernels).